main/src/cgeo/geocaching/connector/su/SuLogin.java

```diff
@@ -0,0 +1,186 @@
+package cgeo.geocaching.connector.su;
+
+import android.app.Application;
+import android.support.annotation.NonNull;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.regex.Pattern;
+
+import cgeo.geocaching.CgeoApplication;
+import cgeo.geocaching.R;
+import cgeo.geocaching.connector.AbstractLogin;
+import cgeo.geocaching.enumerations.StatusCode;
+import cgeo.geocaching.network.Network;
+import cgeo.geocaching.network.OAuth;
+import cgeo.geocaching.network.OAuthTokens;
+import cgeo.geocaching.network.Parameters;
+import cgeo.geocaching.settings.Credentials;
+import cgeo.geocaching.settings.Settings;
+import cgeo.geocaching.utils.Log;
+import cgeo.geocaching.utils.MatcherWrapper;
+import okhttp3.Response;
+
+public class SuLogin extends AbstractLogin {
+
+    private static final Pattern PARAMS_PATTERN_1 = Pattern.compile("oauth_token=([\\w_.-]+)");
+    private static final Pattern PARAMS_PATTERN_2 = Pattern.compile("oauth_token_secret=([\\w_.-]+)");
+    private static final String host = "www.geocaching.su";
+    private static final String FULL_HOST_URL = "http://" + host;
```

Comment:
No chance for https? I mean, setting up let's encrypt is supposed to be quite easy.

Reply:
Not so easy. The server and PHP version is super ancient,we have HTTPS in plans hopefully soon.